src/fu-util.c

```diff
@@ -560,6 +617,227 @@ fu_util_install_prepared (FuUtilPrivate *priv, gchar **values, GError **error)
+		json_builder_set_member_name (builder, "Plugin");
+		json_builder_add_string_value (builder, fu_device_get_plugin (dev));
+
+		/* report what we're trying to update *from* and *to* */
+		json_builder_set_member_name (builder, "Version");
+		json_builder_add_string_value (builder, fu_device_get_version (dev));
+		json_builder_set_member_name (builder, "VersionNew");
+		json_builder_add_string_value (builder, fu_device_get_version_new (dev));
+
+		/* to know the state of the dev we're trying to update */
+		json_builder_set_member_name (builder, "Flags");
+		json_builder_add_int_value (builder, fu_device_get_flags (dev));
+
+		/* to know when the update tried to happen, and how soon after boot */
+		json_builder_set_member_name (builder, "Created");
+		json_builder_add_int_value (builder, fu_device_get_created (dev));
+		json_builder_set_member_name (builder, "Modified");
+		json_builder_add_int_value (builder, fu_device_get_modified (dev));
+
+		json_builder_end_object (builder);
+	}
+	json_builder_end_array (builder);
+	json_builder_end_object (builder);
+
+	/* export as a string */
+	json_root = json_builder_get_root (builder);
+	json_generator = json_generator_new ();
+	json_generator_set_pretty (json_generator, TRUE);
+	json_generator_set_root (json_generator, json_root);
+	data = json_generator_to_data (json_generator, NULL);
+	if (data == NULL) {
+		g_set_error_literal (error,
+				     FWUPD_ERROR,
+				     FWUPD_ERROR_INTERNAL,
+				     "Failed to convert to JSON string");
+		return FALSE;
+	}
+
+	/* ask for permission */
+	fu_util_print_data (_("Target"), report_uri);
+	fu_util_print_data (_("Payload"), data);
+	g_print ("%s [Y|n]: ", _("Proceed with upload?"));
+	if (!fu_util_prompt_for_boolean (TRUE)) {
+		g_set_error_literal (error,
+				     FWUPD_ERROR,
+				     FWUPD_ERROR_PERMISSION_DENIED,
+				     "User declined action");
+		return FALSE;
+	}
+
+	/* POST request */
+	msg = soup_message_new (SOUP_METHOD_POST, report_uri);
+	soup_message_set_request (msg, "application/json; charset=utf-8",
+				  SOUP_MEMORY_COPY, data, strlen (data));
+	status_code = soup_session_send_message (priv->soup_session, msg);
+	if (!SOUP_STATUS_IS_SUCCESSFUL (status_code)) {
+		g_set_error (error,
+			     FWUPD_ERROR,
+			     FWUPD_ERROR_INVALID_FILE,
+			     "Failed to upload to %s: %s",
+			     report_uri, soup_status_get_phrase (status_code));
+		return FALSE;
+	}
+
+	/* mark each device as reported */
+	for (guint i = 0; i < results->len; i++) {
+		FuDevice *dev = g_ptr_array_index (results, i);
+		if (fu_device_has_flag (dev, FWUPD_DEVICE_FLAG_REPORTED))
+			continue;
+		g_debug ("setting flag on %s", fu_device_get_id (dev));
+		if (!fwupd_client_report (priv->client,
+					  fu_device_get_id (dev),
+					  NULL, error))
+			return FALSE;
+	}
+
+	return TRUE;
+}
+
+static gboolean
+fu_util_history (FuUtilPrivate *priv, gchar **values, GError **error)
+{
+	g_autoptr(GPtrArray) results = NULL;
+
+	/* get all updates */
```

Comment:
I don't think this comment is right.  it's just getting all the devices, not the ones with updates right?

Reply:
Fixed.